custom/enikshay/reports/filters.py

```diff
@@ -47,14 +47,18 @@ def selected(self):
     @classmethod
     def get_value(cls, request, domain):
         selected = super(EnikshayLocationFilter, cls).get_value(request, domain)
-        if not request.couch_user.has_permission(domain, 'access_all_locations'):
-            selected += request.couch_user.get_location_ids(domain)
         choice_provider = LocationChoiceProvider(StubReport(domain=domain), None)
         choice_provider.configure({'include_descendants': True})
-        return [
+        selected_locations = [
             choice.value
             for choice in choice_provider.get_choices_for_known_values(selected, request.couch_user)
         ]
+        if len(selected_locations) == 0 and not request.couch_user.has_permission(domain, 'access_all_locations'):
```

Comment:
nitpick: the idiomatic way to test for list emptiness according to PEP8 is `if not selected_locations and ...`